tools/run_block.py

```diff
@@ -101,15 +102,16 @@ def run_generator(
     else:
         flags = 0
 
-    _, result = block_generator.program.run_with_cost(max_cost, flags, block_generator.generator_refs)
+    args = create_generator_args(block_generator.generator_refs).first()
+    _, block_result = block_generator.program.run_with_cost(max_cost, flags, args)
```

Comment:
I'm not sure I understand why this works actually. Here's my understanding:

When the full node runs block generators it runs them through our ROM program that takes all the referenced generator programs as byte blobs. it will parse them (in chialisp) and pass them on to the actual generator that's in the block. So, there are two levels of the generator interface.

1. going through the ROM program, the inputs are flat buffers of clvm serialized programs
2. going directly to the generator program as it appears in the block, the inputs are the clvm programs themselves, as clvm tokens (not as a serialized buffer).

In my original patch, I was going for (2), just to keep things simple. One observation is that even though our referenced generators are serialized (in `SerializedProgram`) they are treated as the clvm tokens, not as flat buffers. This is because the interface to `clvm_rs` itself passes all buffers in serialized form. So, `generator_refs` will pass the clvm-tokens to the program. As far as I understand, what `create_generator_args()` does is two things:

1. turns the `SerializedProgram` obejct into clvm atoms containing the serialized form (which is the input the ROM program expects)
2. wraps this list in another list, as this API allows easy extensions in the futre.

Given that this code doesn't use the generator ROM code, I think we want to pass in the `SerializedPrograms` in their existing form, i.e. not turn them into atoms of serialized bytes. The thing that was missing originally was to wrap them in an extra list.